crates/ibc/src/core/ics04_channel/handler/recv_packet.rs

```diff
@@ -70,13 +71,8 @@ where
         let client_id_on_b = conn_end_on_b.client_id();
         let client_state_of_a_on_b = ctx_b.client_state(client_id_on_b)?;
 
-        // The client must not be frozen.
-        if client_state_of_a_on_b.is_frozen() {
-            return Err(PacketError::FrozenClient {
-                client_id: client_id_on_b.clone(),
-            }
-            .into());
-        }
+        client_state_of_a_on_b.confirm_not_frozen()?;
```

Comment:
ibc-go only checks if client is frozen for `send_packet` (of all packet-related handlers). Waiting on response from them for why that is

Reply:
wrong; all the handlers do make that check as expected